### PR TITLE
To add the user bin folder to the path

### PR DESCRIPTION
**Pull request by aborruso**
Here the thread https://github.com/jupyterhub/binderhub/issues/557

### CODE REVIEW

File: repo2docker/buildpacks/base.py [hunk 1]
```diff
@@ -71,6 +71,9 @@
 {% endfor -%}
 {% endif -%}
 
+# So users can move executables here with postBuild, issue #557
+ENV PATH {{ /home/jovyan/{local}/bin:${PATH}}}
```
Comment (betatim) [inline repo2docker/buildpacks/base.py]: I think this needs to be `'/home/jovyan/local/bin'` instead of `/home/jovyan/{local}/bin`.

**Comment by betatim** [comment]
Any opinions on `/home/jovyan/bin` vs `/home/jovyan/local/bin`?

**Comment by choldgraf** [comment]
gauging from https://unix.stackexchange.com/a/8658 it does seem like `/usr/local/bin` is better because it's something that the user would be manually adding things to. WDYT?

**Comment by choldgraf** [comment]
looking through it again, I don't think we need the double-curlies in there (and I think that's why a bunch of tests were failing too). I just patched the PR so let's see if that fixes it

**Comment by betatim** [comment]
The SO article you linked talks about `/usr/local/bin` but you changed it to `~/local/bin`. Is that what you meant to do? I don't have any strong feelings for which path we end up using. Potentially a normal user isn't allowed to write to `/usr/local/bin` hence I was thinking use a path in the home directory.

File: repo2docker/buildpacks/base.py [hunk 2]
```diff
@@ -71,6 +71,9 @@
 {% endfor -%}
 {% endif -%}
 
+# So users can move executables here with postBuild, issue #557
```
Comment (betatim) [inline repo2docker/buildpacks/base.py]: I wouldn't refer to the issue (we will forget which repo etc) and instead include any additional info we do need in the comment here. Personally I think the comment is fine as it is.

**Comment by betatim** [comment]
Add a short test for this and then merge?

(restarted travis as it was the R tests that were failing)

**Comment by choldgraf** [comment]
+1 on merge one way or another

@betatim do you think we should add `~/bin` as well? Or keep it as-is?

**Comment by choldgraf** [comment]
woot, thansk for the patch @aborruso !!!

**Comment by aborruso** [comment]
@choldgraf you are very kind, but it is only thanks to you all.

I can not wait to try it on mybinder :)

**Comment by betatim** [comment]
Should be deployed now.

**Comment by aborruso** [comment]
@betatim it works!!!!!!!

Great

**Comment by yuvipanda** [comment]
Congratulations on the PR, @aborruso! 

I've made a minor change to this and proposed it ad #313. It changes the directory from ~/local to ~/.local to be consistent with where other software installs per-user software by default. I'm hoping to get that merged and deployed sooner than latter so not too many repos start depending on it :) Hope that's not too much of a change for you!

**Comment by aborruso** [comment]
thank you @yuvipanda you are right. I will make some small changes in my code ;)

**Pull request merged**